packages/components/focus-trap/src/focus-trap.vue

```diff
@@ -236,14 +276,19 @@ export default defineComponent({
       if (trapContainer) {
         trapContainer.removeEventListener(FOCUS_AFTER_TRAPPED, trapOnFocus)
 
-        const releasedEvent = new Event(
-          FOCUS_AFTER_RELEASED,
-          FOCUS_AFTER_TRAPPED_OPTS
-        )
+        const releasedEvent = new CustomEvent(FOCUS_AFTER_RELEASED, {
+          ...FOCUS_AFTER_TRAPPED_OPTS,
+          detail: {
+            focusReason: focusReason.value,
+          },
+        })
         trapContainer.addEventListener(FOCUS_AFTER_RELEASED, releaseOnFocus)
         trapContainer.dispatchEvent(releasedEvent)
 
-        if (!releasedEvent.defaultPrevented) {
+        if (
+          !releasedEvent.defaultPrevented &&
+          (focusReason.value == 'keyboard' || !isFocusCausedByUserEvent())
```

Comment:
@Giwayume I found that there might be a problem here

Since version 2.2.18, When using the mouse to close the Dialog, the focus will not return to the previous button. It is works normally in 2.2.17. [demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHRleHQgQGNsaWNrPVwiZGlhbG9nVmlzaWJsZSA9IHRydWVcIj5cbiAgICBjbGljayBtZVxuICA8L2VsLWJ1dHRvbj5cbiAgPGVsLWRpYWxvZyB2LW1vZGVsPVwiZGlhbG9nVmlzaWJsZVwiPlxuICAgIDxzcGFuPkNsaWNrIHRoZSBDb25maXJtIGJ1dHRvbiB3aXRoIHRoZSBtb3VzZTwvc3Bhbj5cbiAgICA8dGVtcGxhdGUgI2Zvb3Rlcj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGlhbG9nLWZvb3RlclwiPlxuICAgICAgICA8ZWwtYnV0dG9uIEBjbGljaz1cImRpYWxvZ1Zpc2libGUgPSBmYWxzZVwiPkNhbmNlbDwvZWwtYnV0dG9uPlxuICAgICAgICA8ZWwtYnV0dG9uIHR5cGU9XCJwcmltYXJ5XCIgQGNsaWNrPVwiZGlhbG9nVmlzaWJsZSA9IGZhbHNlXCI+XG4gICAgICAgICAgQ29uZmlybVxuICAgICAgICA8L2VsLWJ1dHRvbj5cbiAgICAgIDwvc3Bhbj5cbiAgICA8L3RlbXBsYXRlPlxuICA8L2VsLWRpYWxvZz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IGRpYWxvZ1Zpc2libGUgPSByZWYoZmFsc2UpXG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

Reply:
I see, it's a problem in Chrome, not in Firefox.

Reply:
@tolking 大致测了下，如果这个地方加上`focusReason.value == 'pointer'` https://github.com/element-plus/element-plus/pull/9941#issuecomment-1259694187的问题重现。
![图片](https://user-images.githubusercontent.com/101238421/218670900-1bf8af21-c86f-406a-b8fb-521bb8d3ab22.png)

Reply:
> I see, it's a problem in Chrome, not in Firefox.

I also repeated this problem on Firefox (MacOS 13.2, Firefox 109.0.1)